backend/lib/richard_burton/publication_index_flat_publication.ex

```diff
@@ -0,0 +1,37 @@
+defmodule RichardBurton.Publication.Index.FlatPublication do
+  @moduledoc """
+  Schema for publications
+  """
+  use Ecto.Schema
+
+  alias RichardBurton.Publication.Index.FlatPublication
+
+  @external_attributes [
+    :title,
+    :year,
+    :country,
+    :publisher,
+    :authors,
+    :original_title,
+    :original_authors
+  ]
+
+  @derive {Jason.Encoder, only: @external_attributes}
```

Comment:
I don't like having that `to_map` function at all, I even consider it technical debt 😂 

`Jason` and `to_map` aren't there to solve the same issue, though in this PR it might look like it.

I've been struggling to consistently handle atom keys vs string keys in maps. Specially when it comes to coding helper functions like those in `Publication.Codec`. I understand the notion that data flowing from the outside world must have string keys, while data generated inside can have atom keys. 

Key design decision here (may be flawed in some sense), is that helpers work mostly on attribute maps instead of structs, so I can easily adapt "atom key code" to be used with string keys. However, "inside data" almost always comes from the database in the form of a struct with lots of extra fields. `to_map` is to abstract from the persistence layer for further processing, when necessary. When no intermediate processing is needed, then I just return the data as structs and let `Jason` do the job, hence the `@derive`.

An alternative is to define `to_map` as something like: `struct |> Jason.encode |> Jason.decode |> to_atom_keys` but I don't think it's the way to go 😅 

Reality is, as I figured this out I decided to roll with the `to_map` and I didn't have time to tackle the refactor, although it bothers me.

I don't love the idea of using structs everywhere, but would it be the correct approach?

Reply:
There's also the point that using some artifact like `to_map` makes it easier to assert with `==` in tests.

Reply:
I carried out a big refactor to tidy this up, available in PR https://github.com/wyeworks/richardburton/pull/93